### PR TITLE
#43 - Expose deadline and cancellationToken in public API and provide…

### PR DESCRIPTION
**Pull request by yoricksmeets**
Added the deadline and cancellationToken to the public API of all the methods in the 'lockClient' and updated the documentation for this methods. 

PR implements the improvements of #43

### CODE REVIEW

File: dotnet-etcd/lockClient.cs [hunk 1]
```diff
@@ -20,8 +21,12 @@ public partial class EtcdClient : IDisposable
         /// lease associate with the owner expires.
         /// </summary>
         /// <param name="name">is the identifier for the distributed shared lock to be acquired.</param>
+        /// <param name="headers">The initial metadata to send with the call. This parameter is optional.</param>
+        /// <param name="deadline">An optional deadline for the call. The call will be cancelled if deadline is hit.</param>
+        /// <param name="cancellationToken">An optional token for canceling the call.</param>
         /// <returns></returns>
-        public LockResponse Lock(string name, Grpc.Core.Metadata headers = null)
+        public LockResponse Lock(string name, Grpc.Core.Metadata headers = null, DateTime? deadline = null,
+            CancellationToken cancellationToken = default(CancellationToken))
```
Comment (shubhamranjan) [inline dotnet-etcd/lockClient.cs:29]: Probably missed using these parameters inside the method.

Comment (yoricksmeets) [reply dotnet-etcd/lockClient.cs:29]: Sorry about that, I have updated the commit

**Comment by yoricksmeets** [comment]
I can also add these changes to the rest of the parts of the etcdclient to keep things consistent over the entire client.

**Comment by shubhamranjan** [comment]
> I can also add these changes to the rest of the parts of the etcdclient to keep things consistent over the entire client.

I will merge this PR atm to get the changes released and if you wish to add to the rest of the client, you can create a new PR, else I will do it as soon as I have the bandwidth to do so. I will leave the issue open till then.

Thanks for the contribution and helping me improve this library 😃

**Pull request merged**